packages/expo-cli/src/commands/generate-module/configureModule.ts

```diff
@@ -159,43 +159,37 @@ async function configureAndroid(
   { javaPackage, jsPackageName, viewManager }: ModuleConfiguration
 ) {
   const androidPath = path.join(modulePath, 'android');
-  const sourceFilesPath = path.join(
-    androidPath,
-    'src',
-    'main',
-    'kotlin',
-    'expo',
-    'modules',
-    'template'
-  );
-  const destinationFilesPath = path.join(
-    androidPath,
-    'src',
-    'main',
-    'kotlin',
-    ...javaPackage.split('.')
-  );
+  const templatePackagePath = path.join('expo', 'modules', 'template');
+  const targetPackagePath = javaPackage
+    .split('.')
+    .reduce((previous, next) => path.join(previous, next), '');
 
   // remove ViewManager from template
   if (!viewManager) {
-    removeFiles(sourceFilesPath, [`ModuleTemplateView.kt`, `ModuleTemplateViewManager.kt`]);
-
-    replaceContent(path.join(sourceFilesPath, 'ModuleTemplatePackage.kt'), packageContent =>
-      packageContent
-        .replace(/(^\s+)+(^.*?){1}createViewManagers[\s\W\w]+?\}/m, '')
-        .replace(/^.*ViewManager$/, '')
-    );
+    removeViewManagers(androidPath);
   }
+  // Map dir -> array of files within
+  const sourceFilesPaths = await allFilesWithPathContaining(androidPath, templatePackagePath);
+
+  const moduleName = jsPackageName.startsWith('Expo') ? jsPackageName.substring(4) : jsPackageName;
 
-  await fse.mkdirp(destinationFilesPath);
-  await fse.copy(sourceFilesPath, destinationFilesPath);
+  //iterate thru dirs
+  for (let sourcePath of Object.keys(sourceFilesPaths)) {
+    const destinationPath = sourcePath.replace(templatePackagePath, targetPackagePath);
+
+    await fse.mkdirp(destinationPath);
+    for (let file of sourceFilesPaths[sourcePath]) {
+      await fse.copy(
+        path.join(sourcePath, file),
+        path.join(destinationPath, file.replace('ModuleTemplate', moduleName))
+      );
+      await fse.remove(path.join(sourcePath, file));
+    }
```

Comment:
```suggestion
  for (let [sourcePath, files] of Object.entries(sourceFilesPaths)) {
    const destinationPath = sourcePath.replace(templatePackagePath, targetPackagePath);

    await fse.mkdirp(destinationPath);
    for (let file of files) {
      await fse.copy(
        path.join(sourcePath, file),
        path.join(destinationPath, file.replace('ModuleTemplate', moduleName))
      );
      await fse.remove(path.join(sourcePath, file));
    }
```
That's my preference 💅 😉